scripts/obtain/dotnet-install.sh

```diff
@@ -267,13 +267,17 @@ get_latest_version_info() {
     local azure_channel=$2
     local normalized_architecture=$3
     
-    local osname=$(get_current_os_name)
+    local osname
+    local version_file_url
     
-    local version_file_url="$azure_feed/$azure_channel/dnvm/latest.$osname.$normalized_architecture.version"
-    say_verbose "get_latest_version_info: latest url: $version_file_url"
-    
-    download $version_file_url
-    return $?
+    if osname=$(get_current_os_name); then
+	version_file_url="$azure_feed/$azure_channel/dnvm/latest.$osname.$normalized_architecture.version"
+	say_verbose "get_latest_version_info: latest url: $version_file_url"
+	download $version_file_url
+    else
+	return 1
```

Comment:
IMHO users will know that  `get_latest_version_info` failed because the _os detection_ failed. from the script output (Arch Linux here):
```
➜ cli git:(HANDLE_OS_DETECTION_ERRORS)$ ./build.sh
dotnet-install: Calling: machine_has curl
dotnet-install: Calling: calculate_vars 
dotnet-install: Calling: get_azure_channel_from_channel preview
dotnet-install: azure_channel=preview
dotnet-install: Calling: get_normalized_architecture_from_architecture <auto>
dotnet-install: Calling: get_machine_architecture 
dotnet-install: Calling: get_normalized_architecture_from_architecture x64
dotnet-install: normalized_architecture=x64
dotnet-install: Calling: get_specific_version_from_version https://dotnetcli.blob.core.windows.net/dotnet preview x64 Latest
dotnet-install: Calling: get_latest_version_info https://dotnetcli.blob.core.windows.net/dotnet preview x64
dotnet-install: Calling: get_current_os_name 
dotnet_install: Error: OS name could not be detected
➜ cli git:(HANDLE_OS_DETECTION_ERRORS)$ 
```


Would you prefer another message in the else branch like:
```
    say_err "get_latest_version_info failed"?
```

I tought this is to much verbosity and will make it harder for the user to detect the root cause.